src/main/java/org/apache/cassandra/sidecar/cluster/SidecarLoadBalancingPolicy.java

```diff
@@ -193,25 +194,33 @@ private void addLocalHostsToSelected(List<Host> sourceHosts)
 
     private synchronized void recalculateSelectedHosts()
     {
-        // Copy the list to allow us to remove hosts as we build the list
-        List<Host> sourceHosts = new ArrayList<>(this.allHosts);
-        addLocalHostsToSelected(sourceHosts);
-        Collections.shuffle(sourceHosts, this.secureRandom);
-        Iterator<Host> hostIterator = sourceHosts.iterator();
-        while (selectedHosts.size() < totalRequestedConnections && hostIterator.hasNext())
+        Map<Boolean, List<Host>> partitionedHosts = allHosts.stream()
+                                                            .collect(Collectors.partitioningBy(
+                                                            host -> localHostAddresses.contains(
+                                                            host.getEndPoint().resolve())));
```

Comment:
I found declare a local variable reads better than the inlined lambda that is formatted to a small block due to indentation. 
```suggestion
        Predicate<Host> hostPartitioner = host -> localHostAddresses.contains(host.getEndPoint().resolve());
        Map<Boolean, List<Host>> partitionedHosts = allHosts.stream()
                                                            .collect(Collectors.partitioningBy(hostPartitioner));
```